packages/serialization/form/src/formSerializationWriter.ts

```diff
@@ -0,0 +1,168 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+import {
+  DateOnly,
+  Duration,
+  Parsable,
+  SerializationWriter,
+  TimeOnly,
+} from "@microsoft/kiota-abstractions";
+
+export class FormSerializationWriter implements SerializationWriter {
+  private readonly writer: string[] = [];
+  private static propertySeparator = `&`;
+  private depth = -1;
+  public onBeforeObjectSerialization: ((value: Parsable) => void) | undefined;
+  public onAfterObjectSerialization: ((value: Parsable) => void) | undefined;
+  public onStartObjectSerialization:
+    | ((value: Parsable, writer: SerializationWriter) => void)
+    | undefined;
+  public writeStringValue = (key?: string, value?: string): void => {
+    key && value && this.writePropertyName(key);
+    value && this.writer.push(`=${encodeURIComponent(value)}`);
+    key && value && this.writer.push(FormSerializationWriter.propertySeparator);
+  };
+  private writePropertyName = (key: string): void => {
+    this.writer.push(encodeURIComponent(key));
+  };
+  public writeBooleanValue = (key?: string, value?: boolean): void => {
+    value !== null &&
+      value !== undefined &&
+      this.writeStringValue(key, `${value}`);
+  };
+  public writeNumberValue = (key?: string, value?: number): void => {
+    value && this.writeStringValue(key, `${value}`);
+  };
+  public writeGuidValue = (key?: string, value?: string): void => {
+    value && this.writeStringValue(key, `${value}`);
```

Comment:
why are we adding the cost of interpolating values that are already strings in here?

Reply:
```suggestion
    value && this.writeStringValue(key, value);
```